src/validation/tests/duplicates_validation_test.rs

```diff
@@ -532,8 +533,131 @@ fn generics_with_duplicate_symbol_dont_err() {
     assert!(diagnostics.is_empty());
 }
 
+#[test]
+fn duplicate_enum_names() {
+    let diagnostics = parse_and_validate_buffered(
+        r"
+        TYPE
+            foo : (a, b);
+            foo : (c, d);
+        END_TYPE
+        ",
+    );
+
+    assert_snapshot!(diagnostics, @r###"
+    error[E004]: foo: Ambiguous datatype.
+      ┌─ <internal>:3:13
+      │
+    3 │             foo : (a, b);
+      │             ^^^ foo: Ambiguous datatype.
+    4 │             foo : (c, d);
+      │             --- see also
+
+    error[E004]: foo: Ambiguous datatype.
+      ┌─ <internal>:4:13
+      │
+    3 │             foo : (a, b);
+      │             --- see also
+    4 │             foo : (c, d);
+      │             ^^^ foo: Ambiguous datatype.
+
+    "###);
+}
+
+#[test]
+fn duplicate_enum_inline_variants() {
+    let diagnostics = parse_and_validate_buffered(
+        r"
+        FUNCTION main : DINT
+            VAR
+                foo : (a, b, c, a);
+            END_VAR
+        END_FUNCTION
+        ",
+    );
+
+    assert_snapshot!(diagnostics, @r###"
+    error[E004]: __main_foo.a: Duplicate symbol.
```

Comment:
It would be nice to use the global context for this validation to get rid of internal type names, but I think this is outside of the scope of this commit

Reply:
I think I'll address this in another issue if it's ok for you